### ISSUE TITLE
@index() defined in entity level doesn't respect name property in @colomn() option

### ISSUE THREAD

**Issue by sevenryze** [typeorm/typeorm#2063]
**Issue type:**

[ ] question
[ ] bug report
[ ] feature request
[*] documentation issue

**Database system/driver:**

[ ] `cordova`
[ ] `mongodb`
[ ] `mssql`
[*] `mysql` / `mariadb`
[ ] `oracle`
[ ] `postgres`
[ ] `sqlite`
[ ] `sqljs`
[ ] `react-native`

**TypeORM version:**

[*] `latest`
[ ] `@next`
[ ] `0.x.x` (or put your version here)

**Steps to reproduce or a small repository showing the problem:**
Another issue related to custom ID. 😢 

when i use the case like:
```js
@Index(["id", "something"])
export abstract class BaseInfo {
  @PrimaryColumn({
    name: "id",
  })
  readonly id_custom: Buffer;
}
```
typeorm gives me that the index name not defined in entity list.  I guess this should be a design intentional case? or we should doc the index name will not respect to the db real column name but only to the entity property name.

IMO, that's a little but confused detail...

<!--
    To answer those questions you need to put "x" inside the square brackets, for example:
    [x] `mysql`
    [ ] `postgres`

    Also, please format your code properly (by taking code blocks into ```ts .... ```)
!>

**Comment by filipjnc** [comment]
I am not sure it will work, but give this a shot:

```
@Index((baseInfo: BaseInfo) => [baseInfo.id_custom, baseInfo.something])
export abstract class BaseInfo {
  @PrimaryColumn({
    name: "id",
  })
  readonly id_custom: Buffer;
}
```

**Comment by sevenryze** [comment]
Seem like the signature of `@Index()` declarator you written is an un-doc api? I cannot find this form, `(entitiy)=>[entity.key, entitiy.key2]`, anywhere in  our website doc.

**Comment by pleerock** [comment]
yes its undocumented and mostly experimental for now. Official way is the way you wrote @sevenryze . 

> entity level doesn't respect name 

index name is not equal to column name. Index name is generated name. If you want to have a custom name you can provide it
